zkevm-circuits/src/rlp_circuit.rs

```diff
@@ -0,0 +1,1705 @@
+
+            cb.require_equal(
+                "is_last == 1 then value_acc == value_rlc",
+                meta.query_advice(rlp_table.value_acc, Rotation::cur()),
+                meta.query_advice(all_bytes_rlc_acc, Rotation::cur()),
+            );
+            cb.require_equal(
+                "is_last == 1 then tag == RlpTxTag::Rlp",
+                meta.query_advice(rlp_table.tag, Rotation::cur()),
+                RlpTxTag::Rlp.expr(),
+            );
+
+            // if data_type::cur == TxHash
+            // - tx_id does not change.
+            // - TxSign rows follow.
+            cb.condition(
+                meta.query_advice(rlp_table.data_type, Rotation::cur()),
+                |cb| {
+                    cb.require_equal(
+                        "tx_id does not change",
+                        meta.query_advice(rlp_table.tx_id, Rotation::cur()),
+                        meta.query_advice(rlp_table.tx_id, Rotation::next()),
+                    );
+                    cb.require_equal(
+                        "TxSign rows follow TxHash rows",
+                        meta.query_advice(rlp_table.data_type, Rotation::next()),
+                        RlpDataType::TxSign.expr(),
+                    );
+                    cb.require_equal(
+                        "TxSign rows' first row is Prefix again",
+                        meta.query_advice(rlp_table.tag, Rotation::next()),
+                        RlpTxTag::Prefix.expr(),
+                    );
+                },
+            );
+
+            // if data_type::cur == TxSign and it was not the last tx in the layout
+            // - tx_id increments.
+            // - TxHash rows follow.
+            cb.condition(
+                and::expr(vec![
+                    not::expr(meta.query_advice(rlp_table.data_type, Rotation::cur())),
+                    meta.query_fixed(q_usable, Rotation::next()),
+                ]),
+                |cb| {
+                    cb.require_equal(
+                        "tx_id increments",
+                        meta.query_advice(rlp_table.tx_id, Rotation::cur()) + 1.expr(),
+                        meta.query_advice(rlp_table.tx_id, Rotation::next()),
+                    );
+                    cb.require_equal(
+                        "TxHash rows follow TxSign rows",
+                        meta.query_advice(rlp_table.data_type, Rotation::next()),
+                        RlpDataType::TxHash.expr(),
+                    );
+                    cb.require_equal(
+                        "TxHash rows' first row is Prefix again",
+                        meta.query_advice(rlp_table.tag, Rotation::next()),
+                        RlpTxTag::Prefix.expr(),
+                    );
+                },
+            );
+
+            cb.gate(and::expr(vec![
+                meta.query_fixed(q_usable, Rotation::cur()),
+                meta.query_advice(is_last, Rotation::cur()),
+            ]))
+        });
+
```

Comment:
Add constraints for padding rows. 

- tag = RlpTxTag::Padding, tag' = RlpTxTag::Padding,
- tx_id = 0, tx_id' = 0
- data_type = 0
- tag_rindex = 0
- value_acc = 0